platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineManager.java

```diff
@@ -119,11 +120,12 @@ public MachineImpl create(final RecipeMachineCreationMetadata machineCreationMet
             throws MachineException, NotFoundException {
         final InstanceProvider instanceProvider = machineInstanceProviders.getProvider(machineCreationMetadata.getType());
         final String recipeType = machineCreationMetadata.getRecipeDescriptor().getType();
-
+        final Recipe recipe = RecipeImpl.fromDescriptor(machineCreationMetadata.getRecipeDescriptor());
```

Comment:
Please replace RecipeDescriptor in RecipeMachineCreationMetadata with MachineRecipe. It allows avoid this code.